ViewModels/DeckViewModel.cs

```diff
@@ -42,73 +37,114 @@ public Deck SelectedDeck
-            _selectedDeck = new Deck
+        private async void OnDeckSelected(IDeck selectedDeck)
+        {
+            if (selectedDeck != null)
             {
-                Name = "Default Deck",
-                Cards = new List<Card>() // Empty card list
-            };
-            ToggleDeckListCommand = new Command(ToggleDeckList);
-            UserDecks = new ObservableCollection<Deck>();
-            SelectedDeckCards = new ObservableCollection<Card>();
-            LoadUserDecks(); // Load user decks
+                try
+                {
+                    Console.WriteLine($"Fetching cards for deck: {selectedDeck.Name}");
+                    var deckCards = await selectedDeck.GetCards();
+                    if (deckCards != null)
+                    {
+                        SelectedDecksCards = new ObservableCollection<ICard>((IEnumerable<ICard>)deckCards);
+                        Console.WriteLine($"Loaded {deckCards.Count} cards for deck: {selectedDeck.Name}");
+                    }
+                    else
+                    {
+                        SelectedDecksCards = new ObservableCollection<ICard>();
+                        Console.WriteLine($"No cards found for deck: {selectedDeck.Name}");
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Error loading deck cards: {ex.Message}");
+                }
+
+            }
         }
 
-        // Load the user's decks from the backend
-        private async Task LoadUserDecks()
+        private void OnCreateDeckCommand(object obj)
         {
-            
+            throw new NotImplementedException();
         }
 
-        private void ToggleDeckList()
+        private async Task InitializeViewModelAsync()
         {
-            IsDeckListVisible = !IsDeckListVisible; // Toggle visibility
+            await LoadUserDecks();
+        }
+
+        // Load the user's decks from the backend
+        public async Task LoadUserDecks()
+        {
+            try
+            {
+                var userDecks = await _backendService.GetDecksAsync();
+
+
+                if (userDecks != null)
+                {
+                    UserDecks = new ObservableCollection<IDeck>(userDecks);
+                    HasDecks = true;
+                    Console.Write($"Had deck is {HasDecks}");
+                }
+                else
+                {
+                    Console.WriteLine("In here; fail");
+                    //User has no decks
+                    HasDecks = false;
+                }
+            }
+            catch (RestException ex)
+            {
+                HasDecks = false;
+                Console.WriteLine(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                HasDecks = false;
+                Console.WriteLine($"General error occurred: {ex.Message}");
+            }
```

Comment:
this can be addressed as a separate issue but we should look into global exception handling instead of catching Generic exception in a bunch of places